fw/http.c

```diff
@@ -3503,6 +3553,36 @@ tfw_h2_adjust_req(TfwHttpReq *req)
 	return r;
 }
 
+/*
+ * Throw away a response body and set "Content-Length" to zero.
+ */
+static int
+tfw_h1_purge_resp_clean(TfwHttpResp *resp)
+{
+	int ret;
+	TfwStr replacement = {
+		.chunks = (TfwStr []) {
+			TFW_STR_STRING("Content-Length"),
+			TFW_STR_STRING(": "),
+			TFW_STR_STRING("0"),
+		},
+		.nchunks = 3,
+	};
+	TfwStr *c = replacement.chunks;
+
+	if (!TFW_STR_EMPTY(&resp->body)) {
+		ret = ss_skb_chop_head_tail(NULL, resp->msg.skb_head, 0,
+					    tfw_str_total_len(&resp->body));
```

Comment:
We also should use skb_head here. Just use `pv` tool to force the kernel to send data with extremely small skbs.

`tfw_str_total_len()` includes EOL - is this `\r\n` just before the body? If yes, then we should not chop it

Reply:
`tfw_str_total_len` here is the body length, it doesn't seem to include any part of the headers block. Then we chop the body length bytes from the end of the request skb list. I think we're good here.